localstack-core/localstack/services/sns/v2/provider.py

```diff
@@ -118,6 +164,360 @@ def set_topic_attributes(
+        self, context: RequestContext, subscription_arn: subscriptionARN, **kwargs
+    ) -> None:
+        if subscription_arn is None:
+            raise InvalidParameterException(
+                "Invalid parameter: SubscriptionArn Reason: no value for required parameter",
+            )
+        count = len(subscription_arn.split(":"))
+        try:
+            parsed_arn = parse_arn(subscription_arn)
+        except InvalidArnException:
+            # TODO: check for invalid SubscriptionGUID
+            raise InvalidParameterException(
+                f"Invalid parameter: SubscriptionArn Reason: An ARN must have at least 6 elements, not {count}"
+            )
+
+        account_id = parsed_arn["account"]
+        region_name = parsed_arn["region"]
+
+        store = self.get_store(account_id=account_id, region=region_name)
+        if count == 6 and subscription_arn not in store.subscriptions:
+            raise InvalidParameterException("Invalid parameter: SubscriptionId")
+
+        # TODO: here was a moto_backend.unsubscribe call, check correct functionality and remove this comment
+        #  before switching to v2 for production
+
+        # pop the subscription at the end, to avoid race condition by iterating over the topic subscriptions
+        subscription = store.subscriptions.get(subscription_arn)
+
+        if not subscription:
+            # unsubscribe is idempotent, so unsubscribing from a non-existing topic does nothing
+            return
+
+        if subscription["Protocol"] in ["http", "https"]:
+            # TODO: actually validate this (re)subscribe behaviour somehow (localhost.run?)
+            #  we might need to save the sub token in the store
+            # TODO: AWS only sends the UnsubscribeConfirmation if the call is unauthenticated or the requester is not
+            #  the owner
+            subscription_token = encode_subscription_token_with_region(region=context.region)
+            message_ctx = SnsMessage(
+                type=SnsMessageType.UnsubscribeConfirmation,
+                token=subscription_token,
+                message=f"You have chosen to deactivate subscription {subscription_arn}.\nTo cancel this operation and restore the subscription, visit the SubscribeURL included in this message.",
+            )
+            publish_ctx = SnsPublishContext(
+                message=message_ctx,
+                store=store,
+                request_headers=context.request.headers,
+                # TODO: add the topic attributes once we ported them from moto to LocalStack
+                # topic_attributes=vars(moto_topic),
+            )
+            self._publisher.publish_to_topic_subscriber(
+                publish_ctx,
+                topic_arn=subscription["TopicArn"],
+                subscription_arn=subscription_arn,
+            )
+
+        with contextlib.suppress(ValueError):
+            store.topic_subscriptions[subscription["TopicArn"]].remove(subscription_arn)
```

Comment:
here this will be adapted to remove it from the `store.topics[subscription["TopicArn"]]["subscriptions"]` instead